packages/core/src/workflows/machine.ts

```diff
@@ -346,15 +346,15 @@ export class Machine<
     return {
       resolverFunction: fromPromise(async ({ input }: { input: ResolverFunctionInput }) => {
         const { stepNode, context } = input;
-        const attemptCount = context.attempts[stepNode.step.id];
+        const attemptCount = context.attempts[stepNode.config.id || stepNode.step.id];
```

Comment:
Can we create a helper for this id fetching? Or a getter on the stepNode object? This way we don't have to duplicate the same logic over and over again.

I'm thinking like this

```suggestion
        const attemptCount = context.attempts[stepNode.id];
```

Stepnode can have a getter function that just returns that expression

Reply:
Good callout 👍
I'll update